### PR TITLE
[CXF-7990]:Fix Infinite loop when service endpoint throws SOAPFaultEx…

### PR DESCRIPTION
**Pull request by jimma**
…ception

### CODE REVIEW

File: systests/jaxws/src/test/java/org/apache/cxf/systest/jaxws/EchoServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import javax.xml.soap.SOAPConstants;
+import javax.xml.soap.SOAPFactory;
+import javax.xml.soap.SOAPFault;
+import javax.xml.ws.soap.SOAPFaultException;
+
+import org.apache.cxf.ext.logging.Logging;
+
+@WebService(name = "MyEchoService", targetNamespace = "urn:echo")
+@Logging
+public class EchoServiceImpl {
```
Comment (coheigea) [inline systests/jaxws/src/test/java/org/apache/cxf/systest/jaxws/EchoServiceImpl.java]: "implements EchoService"

File: core/src/main/java/org/apache/cxf/interceptor/FaultOutInterceptor.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 import java.util.logging.Logger;
 
 import javax.xml.stream.XMLStreamWriter;
+import javax.xml.ws.soap.SOAPFaultException;
```
Comment (dkulp) [inline core/src/main/java/org/apache/cxf/interceptor/FaultOutInterceptor.java]: This makes core depend on SAAJ which is definitely not what we'd want to do.   Let me take a look at the test case to see if there is a different way to handle this from the jaxws fault interceptors.

**Comment by dkulp** [comment]
I believe the better patch would be:

```
--- a/rt/frontend/jaxws/src/main/java/org/apache/cxf/jaxws/interceptors/WebFaultOutInterceptor.java
+++ b/rt/frontend/jaxws/src/main/java/org/apache/cxf/jaxws/interceptors/WebFaultOutInterceptor.java
@@ -181,6 +181,8 @@ public class WebFaultOutInterceptor extends FaultOutInterceptor {
                 //and let the rest of the chain try handling it as is.
                 LOG.log(Level.WARNING, "EXCEPTION_WHILE_WRITING_FAULT", nex);
             }
+        } else if (cause instanceof SOAPFaultException && ((SOAPFaultException)cause).getFault().hasDetail()) {
+            return;
         } else {
             FaultMode mode = message.get(FaultMode.class);
             if (mode == FaultMode.CHECKED_APPLICATION_FAULT) {
```
which moves the check for the detail not the jaxws frontend which is where the handling for the SAAJ based things should be done.

**Comment by jimma** [comment]
Thanks for review @dkulp @coheigea . Change has pushed.

**Pull request merged**